luda-editor/luda-editor-client/src/app/editor/timeline/timeline_body/mod.rs

```diff
@@ -4,18 +4,54 @@ pub mod track_body;
 use super::TimelineRenderContext;
 use crate::app::{
     editor::events::EditorEvent,
-    types::{PixelSize, Track},
+    types::{PixelSize, Time, Track},
 };
 use track_body::*;
 
-pub struct TimelineBody {}
+pub struct TimelineBody {
+    last_clip_clicked_mouse_event_id: Option<String>,
+}
 pub struct TimelineBodyProps<'a> {
     pub width: f32,
     pub height: f32,
     pub tracks: &'a [Arc<Track>],
     pub context: &'a TimelineRenderContext<'a>,
 }
+
+struct TimelineBodyLeftClickEvent {
+    pub mouse_event_id: String,
+    pub mouse_position_in_time: Time,
+}
+
 impl TimelineBody {
+    pub(crate) fn new() -> TimelineBody {
```

Comment:
From now on, TimelineBody take care about selected clip binding click event's id to check whether user clicked empty space of timeline body or not.